cas-server-core-api-ticket/src/main/java/org/jasig/cas/ticket/registry/TicketRegistryState.java

```diff
@@ -15,7 +15,7 @@
      * @return Number of ticket-granting tickets in the registry at time of invocation
      *         or {@link Integer#MIN_VALUE} if unknown.
      */
-    int sessionCount();
+    long sessionCount();
```

Comment:
Well, the fact that the Java API there returns long, does not warrant our
API change, IMHO. I remember specifically casting that count() call to int
to adopt to CAS API. And I'm not sure about billions sessions :-)
On Sat, Mar 5, 2016 at 07:27 Misagh Moayyed <notifications@github.com>
wrote:

> In
> cas-server-core-api-ticket/src/main/java/org/jasig/cas/ticket/registry/TicketRegistryState.java
> <https://github.com/Jasig/cas/pull/1588#discussion_r55120394>:
>
> > @@ -15,7 +15,7 @@
> >       * @return Number of ticket-granting tickets in the registry at time of invocation
> >       *         or {@link Integer#MIN_VALUE} if unknown.
> >       */
> > -    int sessionCount();
> > +    long sessionCount();
>
> Most sessionCount() impls do this:
>
>  getTickets().stream().filter(t -> t instanceof TicketGrantingTicket).count();
>
> that returns a long.
>
> Also, theoretically it's possible for the session count to be a long if
> you have a billion sessions available and thousands and thousands of
> transactions going through. As such, the API here would simply bomb with an
> int.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Jasig/cas/pull/1588/files#r55120394>.
>